src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlRangeTranslator.cs

```diff
@@ -0,0 +1,110 @@
+﻿#region License
+
+// The PostgreSQL License
+//
+// Copyright (C) 2016 The Npgsql Development Team
+//
+// Permission to use, copy, modify, and distribute this software and its
+// documentation for any purpose, without fee, and without a written
+// agreement is hereby granted, provided that the above copyright notice
+// and this paragraph and the following two paragraphs appear in all copies.
+//
+// IN NO EVENT SHALL THE NPGSQL DEVELOPMENT TEAM BE LIABLE TO ANY PARTY
+// FOR DIRECT, INDIRECT, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES,
+// INCLUDING LOST PROFITS, ARISING OUT OF THE USE OF THIS SOFTWARE AND ITS
+// DOCUMENTATION, EVEN IF THE NPGSQL DEVELOPMENT TEAM HAS BEEN ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//
+// THE NPGSQL DEVELOPMENT TEAM SPECIFICALLY DISCLAIMS ANY WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE. THE SOFTWARE PROVIDED HEREUNDER IS
+// ON AN "AS IS" BASIS, AND THE NPGSQL DEVELOPMENT TEAM HAS NO OBLIGATIONS
+// TO PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+
+#endregion
+
+using System;
+using System.Linq.Expressions;
+using JetBrains.Annotations;
+using Microsoft.EntityFrameworkCore.Query.ExpressionTranslators;
+using Npgsql.EntityFrameworkCore.PostgreSQL.Query.Expressions.Internal;
+
+namespace Npgsql.EntityFrameworkCore.PostgreSQL.Query.ExpressionTranslators.Internal
+{
+    /// <summary>
+    /// Provides translation services for PostgreSQL range operators.
+    /// </summary>
+    /// <remarks>
+    /// See: https://www.postgresql.org/docs/current/static/functions-range.html
+    /// </remarks>
+    public class NpgsqlRangeTranslator : IMethodCallTranslator
+    {
+        /// <inheritdoc />
+        [CanBeNull]
+        public Expression Translate(MethodCallExpression methodCallExpression) =>
+            TryTranslateOperator(methodCallExpression);
+
+        /// <summary>
+        /// Attempts to translate the <see cref="MethodCallExpression"/> as a PostgreSQL range operator.
+        /// </summary>
+        /// <param name="expression">The <see cref="MethodCallExpression"/> to be translated.</param>
+        /// <returns>
+        /// The expression if successful; otherwise, null.
+        /// </returns>
+        [CanBeNull]
+        static Expression TryTranslateOperator([NotNull] MethodCallExpression expression)
```

Comment:
Where not merge `TryTranslateOperator` with `Translate` above, to just have the entire switch in `Translate()`? The separation doesn't seem to add any value. In addition, methods start with Try in C# normally return a bool (success/failure) and have an `out` var, so this method isn't really following standard naming.